src/librustdoc/formats/cache.rs

```diff
@@ -464,7 +464,7 @@ impl<'a, 'tcx> DocFolder for CacheBuilder<'a, 'tcx> {
             let impl_item = Impl { impl_item: item };
             if impl_item.trait_did().map_or(true, |d| self.cache.traits.contains_key(&d)) {
                 for did in dids {
-                    self.cache.impls.entry(did).or_insert(vec![]).push(impl_item.clone());
+                    self.cache.impls.entry(did).or_insert_with(Vec::new).push(impl_item.clone());
                 }
```

Comment:
This also seems like churn.

Reply:
Not entirely. At least we don't allocate a `Vec` (even if it doesn't take much space). Without clippy, I wouldn't have cared but since it showed it to me, why not simply fixing it...

Reply:
`vec![]` does not allocate.

Reply:
It creates a `Vec` type which is 8 bytes? Well, it's completely anecdotic but in this case I still consider this as better code.